linux/devdoc/sync_linux_requirements.md

```diff
@@ -0,0 +1,46 @@
+# sync linux
+================
+
+## Overview
+
+`sync linux` is the Linux implementation of the `sync` header using [futex](https://www.man7.org/linux/man-pages/man2/futex.2.html).
+
+## Exposed API
+
+```c
+MOCKABLE_FUNCTION(, bool, wait_on_address, volatile_atomic int32_t*, address, int32_t*, compare_address, uint32_t, timeout_ms);
+MOCKABLE_FUNCTION(, void, wake_by_address_all, void*, address);
+MOCKABLE_FUNCTION(, void, wake_by_address_single, void*, address);
+```
+
+## wait_on_address
+
+```c
+MOCKABLE_FUNCTION(, bool, wait_on_address, volatile_atomic int32_t*, address, int32_t*, compare_address, uint32_t, timeout_ms)
+```
+
+**SRS_SYNC_LINUX_43_001: [** `wait_on_address` shall initialize a `timespec` struct with `.tv_nsec` equal to `timeout* 10^6`. **]**
+
+
+**SRS_SYNC_LINUX_43_002: [** `wait_on_address` shall call `syscall` from `sys/syscall.h` with arguments `FUTEX_PRIVATE_FLAG`, `address`, `FUTEX_WAIT`, `*compare_address`, `timeout_struct`, `NULL`, `NULL`. **]**

```

Comment:
>FUTEX_PRIVATE_FLAG [](start = 105, length = 18)

I think I put the comment on the wrong spot.
We do want `SYS_futex`, but the argument `FUTEXT_WAIT` should be `FUTEX_WAIT|FUTEX_PRIVATE_FLAG`.

That private flag also belongs in the calls below